### PR TITLE
fix(dialog): Closing Animations not running #433

### LINKED ISSUE #433: Dialog Closing Animations not running
## What MDC-Web Version are you using?

0.7.0
## What browser(s) is this bug affecting?

Chrome Version 57.0.2987.110 (32-bit)
Firefox 52.0.1 (64-bit)
## What OS are you using?

Windows 10
## What are the steps to reproduce the bug?

On your Demo site open a Dialog and then Close it.
## What is the expected behavior?

The Dialog should run its "Closing" Animation.
## What is the actual behavior?

The Close method "hides" the Dialog without waiting for the animation to run.

## Any other information you believe would be useful?

The foundation needs to wait for a TransitionEnd event before hiding the the Dialog and closing everything down. At the moment the adapter doesn't allow for a TransitionEnd event.

### PR DESCRIPTION
**Pull request by tranrate**
-  Refactor scss to include Animating class.
-  Modify foundation to use Animating class and transitionEnd event for opening and closing Dialog. Alter open close and destroy methods appropriately.
-  Move style="visibility:hidden" attribute into scss so that it is not required in html.

Closes #433

BREAKING CHANGE: 
-  Dialogs do not require a style="visibility:hidden" attribute in html.
-  registerTransitionEndHandler, deregisterTransitionEndHandler, and isDialog methods must be implemented by the adapter

### CODE REVIEW

File: packages/mdc-dialog/foundation.js [hunk 1]
```diff
@@ -118,6 +128,18 @@ export default class MDCDialogFoundation extends MDCFoundation {
     }
   }
 
+  handleTransitionEnd_(ev) {
```
Comment (amsheehan) [inline packages/mdc-dialog/foundation.js]: Ultra-nit: We usually call this `evt`

File: packages/mdc-dialog/mdc-dialog.scss [hunk 2]
```diff
@@ -30,10 +30,29 @@ $mdc-dialog-dark-theme-bg-color: #303030 !default;
...
+  top: 0;
+  left: 0;
   align-items: center;
   justify-content: center;
-  opacity: 0;
-  z-index: -1;
+  width: 100%;
+  height: 100%;
+  visibility: hidden;
+  z-index: 2;
```
Comment (amsheehan) [inline packages/mdc-dialog/mdc-dialog.scss]: `z-index` is set on line 54 as well. I believe this one is extraneous.

File: packages/mdc-dialog/mdc-dialog.scss [hunk 3]
```diff
@@ -131,7 +133,6 @@ $mdc-dialog-dark-theme-bg-color: #303030 !default;
     padding: 8px;
   }
 
-  // TODO: Replace with breakpoint variable
```
Comment (amsheehan) [inline packages/mdc-dialog/mdc-dialog.scss]: We would like to keep this comment in for now. `//` style comments are stripped out in the compiled CSS, so this only really exists for us in dev. These will all be removed once #23 lands.

**Comment by tranrate** [comment]
Hi @amsheehan 

1.  Demo updated - conflicts with Revamp Demo/Catalog #594 resolved.
2.  Foundation.js line 131 - ev changed to evt.
3.  Mdc-dialog.scss line 134 - Comment added back in.
4.  Mdc-dialog.scss line 40 - z-index left as it is, the second one is for the opaque backdrop which has to be behind the dialog box but in front of the main screen. Used a relative index for the backdrop so that only the main one needs changing if anyone needs to adjust the z-index in the future.

Martin

**Comment by traviskaufman** [comment]
@amsheehan could we get another review here?

Also note that in the commit message body we need to change `BREAKING CHANGES:` to `BREAKING CHANGE:`

**Comment by tranrate** [comment]
@amsheehan 

Commit message changed.

Martin

**Pull request merged**